### ISSUE TITLE
Cannot grab teleport scrolls from stash during a teleport from fountain: "In use"

### ISSUE THREAD

**Issue by Fjara-h** [ValveSoftware/Dota2-Gameplay#5354]
### Item name

Teleport scroll

### Description

When teleporting out of fountain (in range of the shop/stash) it is not possible to grab teleports from stash. I assume this has something to do with their cooldowns being different, but it is not really something that should exist. I don't know what the best solution to it is, but denying the person from collecting from stash is poor design and frustrating.

https://clips.twitch.tv/CrazySuccessfulGoblinPoooound-G9GDQnYuvoVAkxtn

### Example Match ID and Timestamp

6858255575 - 12:05  (viper)

**Comment by commentator24** [comment]
as far as i know buying tp during death, buying tp during tp elsewhere from base, and even buying tp when tp on cooldown, it will automatically stack and won't go to stash. perhaps the items were in courier being delivered back to base? hence it goes to stash. and grabbing teleports from stash during tp should be possible with courier. 

i think clicking grab from stash doesnt work because tp scrolls are not allowed in inventory anymore (special 1 slot). hence cannot combine while in use..
or because free TP scrolls and bought ones are actually 2 different items (different resell value).
otherwise maybe ppl can abuse during tp e.g. keep 1 in stash, resell during tp when it combine to 2 or something.

but yes hope they fix it

**Comment by Fjara-h** [comment]
If you go back to game clock of 12:00 on the replay or 03:13:27 of that VOD, the courier returns a tp to stash while viper is dead. It goes off cooldown, and when grabbing from stash it only grabs the other items in stash. This initial instance does not occur while teleporting, only after the teleport started was the grab all clicked a second time. The only thing that makes sense to me is that it is because of the free tp you get on death.

**Comment by commentator24** [comment]
i watched the VOD a bit..
i'm not sure what you meant by this initial instance does not occur while teleporting, because at 12:08 viper did not actually grab items from stash. this is auto combine for recipes, and hence TP scroll still in stash.
i think they should make it so that it auto combine tp scrolls as well, since tp scrolls can no longer be shared.

the only time viper try to grab items from stash was when TP scroll was in use, which i guess not possible because it's in use..
if viper grab items from stash before tp used, tp scrolls would stack, even when TP scrolls in cooldown. u can test this in lobby.
as i've said before, i think 'in use' message is to prevent abuse reselling tp while in use, or because tp scrolls are in special 1 slot only now

**Comment by Fjara-h** [comment]
Ok, good call. I don't think I've ever actually paid attention to how returning items via courier doesn't auto-combine on return, but on spawn. I've only ever grabbed stash myself when anything was in it and I've never let it automatically do anything (because what else am I going to do while dead)

**Comment by TheUnholyWrath** [comment]
#629

**Comment by Fjara-h** [comment]
Duplicate of #629
